### ISSUE TITLE
Rename 'job*' keys in transform loader to 'task'

### ISSUE THREAD

**Issue by ahal** [taskcluster/taskgraph#25]
The fact we use 'job' everywhere in the TransformLoader (`jobs`, `job-defaults`, `jobs-from`) is an artifact from the old buildbot / tbpl days. It's confusing since `task` is very much the standard terminology in Taskcluster land. Let's rename these and deprecate the term `job`.

One path forward here might be to:
1. Land a patch that supports both `job` and `task`, but if `job` is used, log a `DeprecationWarning`.
2. Remove support for `job` in a future major release.

**Comment by ahal** [comment]
Yeah, I'm aware of the original intent there and it was a decent idea. But in hindsight I think mixing the nomenclature has ended up being more confusing not less. During Taskgraph office hours and in discussions with various teams back in Jan / Feb, one common question was "Why is it called 'jobs' here and 'tasks' everywhere else?".

Personally the distinction between whether a task definition conforms to the Taskcluster spec or not, isn't very useful to me. Since this can easily be determined by whether or not we've reached the end of the `full_task_set` phase.

Either way, there's no rush to land this and I'm happy to discuss more.

**Comment by ahal** [comment]
Also, I'd prefer if the `job` transforms were renamed to `run` or something similar, but maybe that's a separate conversation.

**Comment by escapewindow** [comment]
Ok. I think this is large enough that it's worth a discussion and/or RFC. It's quite possible no one will object and we land this patch. It's also possible that the concept of `jobs`, e.g. the `l10n repack jobs`, as a concept is useful to one or more people, and those fan out to 0-to-many chunked `tasks`. (As opposed to "I defined a single task in my kind.yml; why did I end up with 50? 0?")

If we have that discussion and decide to change it, I'm ok with the change. I mainly wanted to avoid making the change without discussion.

**Comment by ahal** [comment]
Sure, I don't think this warrants an RFC but I'll post the link to this issue and folks can chime in if they like.

**Comment by ahal** [comment]
I'm going to be releasing Taskgraph 2.0 soon, so want to make a decision here. I've posted this issue to various channels and asked some folks and no one seems to opinions besides "sounds good". Also showed this to Johan and he agrees this is a good change. So barring objections I'd like to try and get this landed.

Just rebased and updated the associated PR.

**Comment by JohanLorenzo** [comment]
Hey there! I confirm I'm on board with this. I actually came with up with almost the same conclusion when I forked taskgraph to [jobgraph](https://github.com/jobgraph-dev/jobgraph). There, I went the other way around, I renamed everything `task` into `job`.

Plus, I've always been confused when I had to explain the difference between a `job` and a `task` in Taskgraph to people not familiar with it. Taskgraph already has some complex concepts to grasp first. That's what led me to write this extensive blog post[1]. To me, it would make things a little bit easier if we make the terminology of taskgraph closer to the one in Taskcluter. 

[1] https://johanlorenzo.github.io/blog/2019/10/24/taskgraph-is-now-deployed-to-the-biggest-mozilla-mobile-projects.html
